lib/record_store/provider/ns1.rb

```diff
@@ -0,0 +1,192 @@
+
+        if pruned_answers.empty?
+          client.delete_record(
+            zone: zone,
+            fqdn: record_fqdn,
+            type: record.type
+          )
+          return
+        end
+
+        client.modify_record(
+          zone: zone,
+          fqdn: record_fqdn,
+          type: record.type,
+          params: { answers: pruned_answers }
+        )
+      end
+
+      # Updates an existing record in the zone. It is expected this call modifies external state.
+      #
+      # Arguments:
+      # id - provider specific ID of record to update
+      # record - a kind of `Record` which the record with `id` should be updated to
+      def update(id, record, zone)
+        super # may not need to be implemented?        
+      end
+
+      def build_from_api(api_record, zone)
+        fqdn = Record.ensure_ends_with_dot(api_record["domain"])
+
+        record_type = api_record["type"]
+ 
+        api_record["short_answers"].map do |answer|
+          record = {
+            ttl: api_record["ttl"],
+            fqdn: fqdn.downcase,
+          }
+
+          return if record_type == 'SOA'
+
+          case record_type
+          when 'A', 'AAAA'
+            record.merge!(address: answer)
+          when 'ALIAS'
+            record.merge!(alias: answer)
+          when 'CAA'
+            flags, tag, value = answer.split(' ')
+
+            record.merge!(
+              flags: flags.to_i,
+              tag: tag,
+              value: Record.unquote(value),
+            )
+          when 'CNAME'
+            record.merge!(cname: answer)
+          when 'MX'
+
+            preference, exchange = answer.split(' ')
+
+            record.merge!(
+              preference: preference.to_i, 
+              exchange: exchange,
+            )
+          when 'NS'
+            record.merge!(nsdname: answer)
+          when 'SPF', 'TXT'
+            record.merge!(txtdata: Record.unescape(answer).gsub(';', '\;'))
+          when 'SRV'
+            priority, weight, port, host = answer.split(' ')
+
+            record.merge!(
+              priority: priority.to_i,
+              weight: weight.to_i,
+              port: port.to_i,
+              target: Record.ensure_ends_with_dot(host),
+            )
+          end
+          begin
+            Record.const_get(record_type).new(record)
+          rescue NoMethodError => e
+            raise e
+          end
+        end
+      end
+
+      def build_api_answer_from_record(record)
+        if record.is_a? Record::MX
+          [ record.preference, record.exchange ]
+        elsif record.is_a? Record::TXT or record.is_a? Record::SPF
+          [ record.txtdata ]
+        elsif record.is_a? Record::CAA
+          [ record.flags, record.tag, record.value ]
+        elsif record.is_a? Record::SRV
+          [ record.priority, record.weight, record.port, record.target ]
+        else
+          [ record.rdata_txt ]
+        end
+      end
+
+      def symbolize_keys(hash)
```

Comment:
Don't we have that at a more global project scope ? If not, it might be interesting to have a `utils.rb` file with a `Utils` class that includes that ?

Reply:
I am not 100% getting what the `global project scope` means here. Could you give me more details?

Reply:
There's a symbolize_keys in [ActiveSupport](http://api.rubyonrails.org/classes/Hash.html#method-i-symbolize_keys), however it doesn't feel right pulling in the entire library for one method.  We didn't see any similar methods in RecordStore, so decided to implement it, and keep it in the smallest scope that encompasses all of its uses.  If it ends up needing to be used elsewhere, it can moved to a larger scope.